src/cdk/v2/destinations/the_trade_desk/utils.js

```diff
@@ -26,82 +33,220 @@ const getSignatureHeader = (request, secretKey) => {
   return base;
 };
 
-const responseBuilder = (items, config) => {
-  const { advertiserId, dataServer } = config;
+const prepareFromConfig = (destination) => ({
+  tracker_id: destination.Config?.trackerId,
+  adv: destination.Config?.advertiserId,
+});
+
+const getRevenue = (message) => {
+  const { event, properties } = message;
+  let revenue = properties?.value;
+  const eventsMapInfo = ECOMM_EVENT_MAP[event.toLowerCase()];
+  if (eventsMapInfo?.rootLevelPriceSupported) {
+    const { price, quantity = 1 } = properties;
+    if (price && !Number.isNaN(parseFloat(price)) && !Number.isNaN(parseInt(quantity, 10))) {
+      revenue = parseFloat(price) * parseInt(quantity, 10);
+    }
+  } else if (eventsMapInfo?.revenueFieldSupported) {
+    revenue = properties?.revenue || revenue;
+    if (event.toLowerCase() === 'order completed' && !revenue) {
+      throw new InstrumentationError('value is required for `Order Completed` event');
+    }
+  }
 
-  const payload = { DataProviderId: DATA_PROVIDER_ID, AdvertiserId: advertiserId, Items: items };
+  return revenue;
+};
 
-  const response = defaultRequestConfig();
-  response.endpoint = getFirstPartyEndpoint(dataServer);
-  response.method = defaultPostRequestConfig.requestMethod;
-  response.body.JSON = removeUndefinedAndNullValues(payload);
-  return response;
+const prepareItemsFromProperties = (message) => {
+  const { properties } = message;
+  const items = [];
+  const item = constructPayload(properties, ITEM_CONFIGS);
+  // extractCustomFields(message, properties, ['properties'], ITEM_EXCLUSION_LIST);
+  items.push(item);
+  return items;
 };
 
-const batchResponseBuilder = (items, config) => {
-  const response = [];
-  const itemsChunks = BatchUtils.chunkArrayBySizeAndLength(items, {
-    // TODO: use destructuring at the top of file once proper 'mocking' is implemented.
-    // eslint-disable-next-line unicorn/consistent-destructuring
-    maxSizeInBytes: tradeDeskConfig.MAX_REQUEST_SIZE_IN_BYTES,
+const prepareItemsFromProducts = (message) => {
+  const products = get(message, 'properties.products');
+  const items = [];
+  products.forEach((product) => {
+    const item = constructPayload(product, ITEM_CONFIGS);
+    extractCustomFields(product, item, 'root', ITEM_EXCLUSION_LIST);
```

Comment:
Second variable for `extractCustomFields` function definition is misleading. https://github.com/rudderlabs/rudder-transformer/blob/develop/src/v0/util/index.js#L1311
Rename it from `destination` to `payload`
cc: @koladilip @utsabc

Reply:
Renamed **destination** to **payload**

Reply:
Highlight this change in PR description if you are doing in same PR

Reply:
added in PR description